tests/transforms/test_batch_transform.py

```diff
@@ -0,0 +1,427 @@
+            qml.RX(x, wires=0)
+            qml.expval(qml.PauliX(0))
+
+        tapes, fn = my_transform(tape, a, b)
+
+        assert len(tapes[0].operations) == 2
+        assert tapes[0].operations[0].name == "Hadamard"
+        assert tapes[0].operations[1].name == "RY"
+        assert tapes[0].operations[1].parameters == [a * np.abs(x)]
+
+        assert len(tapes[1].operations) == 2
+        assert tapes[1].operations[0].name == "Hadamard"
+        assert tapes[1].operations[1].name == "RZ"
+        assert tapes[1].operations[1].parameters == [b * np.sin(x)]
+
+    def test_parametrized_transform_qnode(self, mocker):
+        """Test that a parametrized transform can be applied
+        to a QNode"""
+
+        @qml.batch_transform
+        def my_transform(tape, a, b):
+            """Generates two tapes, one with all RX replaced with RY,
+            and the other with all RX replaced with RZ."""
+
+            tape1 = qml.tape.QuantumTape()
+            tape2 = qml.tape.QuantumTape()
+
+            # loop through all operations on the input tape
+            for op in tape.operations + tape.measurements:
+                if op.name == "RX":
+                    wires = op.wires
+                    param = op.parameters[0]
+
+                    with tape1:
+                        qml.RY(a * qml.math.abs(param), wires=wires)
+
+                    with tape2:
+                        qml.RZ(b * qml.math.sin(param), wires=wires)
+                else:
+                    for t in [tape1, tape2]:
+                        with t:
+                            qml.apply(op)
+
+            def processing_fn(results):
+                return qml.math.sum(qml.math.stack(results))
+
+            return [tape1, tape2], processing_fn
+
+        a = 0.1
+        b = 0.4
+        x = 0.543
+
+        dev = qml.device("default.qubit", wires=2)
+
+        @qml.qnode(dev)
+        def circuit(x):
+            qml.Hadamard(wires=0)
+            qml.RX(x, wires=0)
+            return qml.expval(qml.PauliX(0))
+
+        transform_fn = my_transform(circuit, a, b)
+
+        spy = mocker.spy(my_transform, "construct")
+        res = transform_fn(x)
+
+        spy.assert_called()
+        tapes, fn = spy.spy_return
+
+        assert len(tapes[0].operations) == 2
+        assert tapes[0].operations[0].name == "Hadamard"
+        assert tapes[0].operations[1].name == "RY"
+        assert tapes[0].operations[1].parameters == [a * np.abs(x)]
+
+        assert len(tapes[1].operations) == 2
+        assert tapes[1].operations[0].name == "Hadamard"
+        assert tapes[1].operations[1].name == "RZ"
+        assert tapes[1].operations[1].parameters == [b * np.sin(x)]
+
+        expected = fn(dev.batch_execute(tapes))
+        assert res == expected
+
+    def test_parametrized_transform_qnode_decorator(self, mocker):
+        """Test that a parametrized transform can be applied
+        to a QNode as a decorator"""
+
+        @qml.batch_transform

```

Comment:
This same transform is used in a couple places; perhaps make it global to the file?

Reply:
Good idea!

Reply:
done!